crates/core/src/client/tests/mod.rs

```diff
@@ -124,3 +136,62 @@ async fn test_transaction_by_hash() {
     assert_eq!(*ABDEL_ETHEREUM_ADDRESS, tx.from);
     assert_eq!(U256::from(0), tx.nonce);
 }
+
+#[tokio::test]
+async fn test_simulate_transaction() {
+    // Given
+    let client = init_testnet_client();
+    let block_id = StarknetBlockId::Tag(BlockTag::Latest);
+    let rpc_client = JsonRpcClient::new(HttpTransport::new(
+        Url::from_str("https://starknet-goerli.infura.io/v3/f453f09c949a453c9bc03a34efcca010").unwrap(),
+    ));
+
+    let block_number = rpc_client.block_number().await.unwrap();
+    let nonce = rpc_client.get_nonce(block_id, *ACCOUNT_ADDRESS).await.unwrap();
+
+    let calldata = vec![
+        FieldElement::ONE,  // call array length
+        *COUNTER_ADDRESS,   // counter address
```

Comment:
how do we know counter is deployed on testnet

Reply:
I could make a call and check if the contract if deployed before the test, or would you prefer to make a call AND deploy if the contract is not found?